pallets/loans/src/entities/interest.rs

```diff
@@ -0,0 +1,98 @@
+use cfg_traits::{InterestAccrual, RateCollection};
+use cfg_types::adjustments::Adjustment;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{traits::UnixTime, RuntimeDebugNoBound};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{EnsureAdd, EnsureSub, Zero},
+	DispatchError, DispatchResult,
+};
+
+use crate::pallet::Config;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebugNoBound, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+pub struct ActiveInterestRate<T: Config> {
+	/// The current interest rate value (per year).
+	/// It the rate it has been penalized,
+	/// it contains the result of applying that penalty:
+	/// rate = base_rate + penalty
+	rate: T::Rate,
+
+	/// Normalized accumulation of the interest rate.
+	/// Used to get the current interest per second
+	normalized_acc: T::Balance,
+
+	/// Penalty applied to this interest rate
+	penalty: T::Rate,
+}
+
+impl<T: Config> ActiveInterestRate<T> {
+	pub fn activate(interest_rate: T::Rate) -> Result<Self, DispatchError> {
+		T::InterestAccrual::reference_rate(interest_rate)?;
+		Ok(Self {
+			rate: interest_rate,
+			normalized_acc: T::Balance::zero(),
+			penalty: T::Rate::zero(),
+		})
+	}
+
+	pub fn deactivate(self) -> Result<T::Rate, DispatchError> {
+		T::InterestAccrual::unreference_rate(self.rate)?;
+		Ok(self.rate)
+	}
+
+	pub fn has_debt(&self) -> bool {
+		!self.normalized_acc.is_zero()
+	}
+
+	pub fn rate(&self) -> T::Rate {
+		self.rate
+	}
+
+	pub fn penalty(&self) -> T::Rate {
+		self.penalty
+	}
+
+	pub fn current_debt(&self) -> Result<T::Balance, DispatchError> {
+		let now = T::Time::now().as_secs();
+		T::InterestAccrual::calculate_debt(self.rate, self.normalized_acc, now)
+	}
+
+	pub fn current_debt_cached<Rates>(&self, cache: &Rates) -> Result<T::Balance, DispatchError>
+	where
+		Rates: RateCollection<T::Rate, T::Balance, T::Balance>,
+	{
+		cache.current_debt(self.rate, self.normalized_acc)
```

Comment:
`RateCollection.current_debt` expects an interest rate per sec, while this is passing an interest rate per year. How does that work? 🤔

Reply:
It actually expects an interest rate per year from #1221. It makes the conversion inside